src/App.tsx

```diff
@@ -15,61 +15,140 @@ const goodsFromServer = [
   'Garlic',
 ];
 
+const preparedGoods = goodsFromServer.map(name => ({
+  name,
+  id: uuidv4(),
+}));
+
 enum SortType {
   NONE,
-  ALPABET,
+  ALPHABET,
   LENGTH,
 }
 
+type Good = {
+  name: string;
+  id: string;
+};
+
 // Use this function in the render method
-function getReorderedGoods(
-  goods: string[],
+const getReorderedGoods = (
+  goods: Good[],
   sortType: SortType,
   isReversed: boolean,
-) {
-  // Not to mutate the original array
+) => {
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // ...
+  if (sortType !== SortType.NONE) {
+    visibleGoods.sort((a, b) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return a.name.localeCompare(b.name);
+        case SortType.LENGTH:
+          return a.name.length - b.name.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-  return visibleGoods;
-}
+  return isReversed ? visibleGoods.reverse() : visibleGoods;
+};
 
-// DON'T save goods to the state
 type State = {
   isStarted: boolean,
   isReversed: boolean,
   sortType: SortType,
 };
 
-export const App = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+export class App extends Component<{}, State> {
+  state = {
+    isStarted: false,
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  sortByName = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  reset = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+  };
+
+  render() {
+    const { isStarted, isReversed, sortType } = this.state;
+    const goods = getReorderedGoods(preparedGoods, sortType, isReversed);
+
+    return (
+      <div className="App level is-flex-direction-column">
+        {!isStarted && (
+          <button
+            type="button"
+            className="button is-success level-item"
+            onClick={() => this.setState({ isStarted: true })}
```

Comment:
u can take it separately

Reply:
It should be specific method, no inliners

Reply:
> u can take it separately



> It should be specific method, no inliners

Ok, thanks, guys)